test-cypress/integration/explore/similarity.spec.js

```diff
@@ -0,0 +1,299 @@
+        cy.get('div[class="card mb-2"]').should('be.visible').should('contain', 'No Indexes');
+    };
+
+	let selectIndexType = function (indexType) {
+        if (indexType == 'predication') {
+            cy.get('#create-predication-index').click();
+        }
+	};
+
+	let cloneExistingIndex = function () {
+        cy.url().should('eq', Cypress.config('baseUrl') + '/similarity');
+        cy.get('i[class="icon-copy"]').click();
+        cy.wait(500);
+        createSimilarityIndex();
+        cy.url().should('contain', Cypress.config('baseUrl') + '/similarity'); //Should change the "contain" method to "eq" once GDB-3699 is resolved
+	};
+
+	let rebuildIndex = function () {
+        cy.url().should('contain', Cypress.config('baseUrl') + '/similarity'); //Should change the "contain" method to "eq" once GDB-3699 is resolved
+        cy.get('span[class="icon-reload"]').click();
+        cy.get('button[class="btn btn-primary"]').click();
+        cy.get('span i[class="icon-reload loader"]').should("be.visible");
+        cy.get('button span[class="icon-trash"]').should("be.visible");
+        cy.get('span i[class="icon-reload loader"]').should("not.be.visible");
+	};
+
+	let viewSPARQLQuery = function () {
+        cy.url().should('contain', Cypress.config('baseUrl') + '/similarity'); //Should change the "contain" method to "eq" once GDB-3699 is resolved
+        cy.get('i[class="icon-sparql"]').click();
+        cy.get('h3[class="modal-title"]').should('be.visible').should('contain', 'View SPARQL Query');
+        cy.get('button[class="btn btn-primary"]').should('be.visible').should('contain', 'Copy to clipboard');
+        cy.get('button[class="close"]').click();
+	};
+
+	let changeDataQuery = function () {
+        cy.url().should('contain', Cypress.config('baseUrl') + '/similarity/index/create'); //Should change the "contain" method to "eq" once GDB-3699 is resolved
+        cy.get('.CodeMirror textarea').type('{ctrl}a{backspace}', {force: true});
+        cy.get('.CodeMirror textarea').type(modifiedDataQuery, {force: true});
+        cy.get('button[class="btn btn-link"]').eq(2).click(); //click test query button to verify results are retrieved
+        cy.get('div a[class="uri"]').should('contain', 'http://dbpedia.org/resource/Aaron_Jay_Kernis');
+
+	};
+
+	let changeSearchQuery = function () {
+        cy.url().should('contain', Cypress.config('baseUrl') + '/similarity/index/create'); //Should change the "contain" method to "eq" once GDB-3699 is resolved
+        cy.get('ul[class="nav nav-tabs"] li[class="nav-item"]').eq(1).click(); //click Search query tab
+        cy.get('.CodeMirror textarea').type('{ctrl}a{backspace}', {force: true}) //delete default search query
+        cy.get('.CodeMirror textarea').invoke('val', modifiedSearchQuery).trigger('change', {force: true});
+        cy.wait(500);
+	};
+
+    it('Test similarity page default state', function () {
+        checkSimilarityPageDefaultState();
+	});
+
```

Comment:
IMO the next tests are not structured properly. There is too many duplication which can be avoided by having them grouped in a spec with proper beforeEach block. Also there are many assertions in the precondition steps which are executed again and again for every test which also slows down the test suite. Better have a test that verifies those things and remove the assertions from these precondition steps.